<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>08_some_questions_1</title>
</head>

<body>
  <script>
    /*
    - promise.then()返回的新promise的结果状态由什么决定？
    简单表达：由then()指定的回调函数执行的结果决定
    详细表达：
     1 如果抛出异常，新promise变为rejected，reason为抛出的异常
     2 如果返回的是非promise的任意值，新promise变为resolved，value为返回的值
     3 如果返回的是另一个新promise，此promise的结果就会成为新promise的结果
    */

    new Promise((resolve, reject) => {
      // resolve(1);
      reject(11)
    }).then(
      value => {
        console.log('onResolved1() ', value)
        // return 2;
        // return Promise.resolve(3);
        // return Promise.reject(4);
        throw 5;
      },
      reason => { // 如果没有把状态改为失败（reject或抛出错误）则默认返回成功状态
        console.log('onRejected1() ', reason) 
        // return Promise.reject(22);
        // return 2;
        // return Promise.resolve(3);
        // return Promise.reject(4);
        throw 5;
      }
    ).then(
      value => {
        console.log('onResolved2() ', value)
      },
      reason => {
        console.log('onRejected2() ', reason)
      }
    )

  </script>
</body>

</html>